.editorconfig

```diff
@@ -126,3 +126,7 @@ dotnet_diagnostic.CA1062.severity = none
 dotnet_diagnostic.CA1303.severity = none
 # Suppressed[CA1001] as not sure how to fix disposable fields within tests
 dotnet_diagnostic.CA1001.severity = none
+dotnet_diagnostic.CA2227.severity = none # list properties should (not?) be readonly
+dotnet_diagnostic.CA2234.severity = none # call HttpClient.PostAsync(Uri, HttpContent)
```

Comment:
Do you have more info about this error?

Reply:
Full error text is `Modify 'HousingInformationGateway.GetResidentInformation(ResidentQueryParam)' to call 'HttpClient.PostAsync(Uri, HttpContent)' instead of 'HttpClient.PostAsync(string, HttpContent)'.`

This is caused by an argument in the gateway which calls `new StringContent()`. We attempted to replace this with `new HttpContent()`, but that threw a different error: `Cannot create an instance of the abstract class or interface 'HttpContent'`

Reply:
I think this is being caused because the URL is being passed as a string, when it wants a URI object. 
So, for example, where you have the code `var response = await _client.PostAsync(_baseUrl, new StringContent(rqpString));`Instead of passing strings,  you can instead create a URI object using `var builder =new UriBuilder(_baseUrl)`, then add the query params using `builder.Query = rqpString` then you can pass `builder.Uri` to the PostAsync method. [here](https://docs.microsoft.com/en-us/dotnet/api/system.uribuilder?view=netcore-3.1) are the docs for the UriBuilder class